routing/service_container_parameters.rst

```diff
@@ -16,6 +16,27 @@ inside your routing configuration:
 
 .. configuration-block::
 
+    .. code-block:: php-annotations
+
+        // src/AppBundle/Controller/MainController.php
+        namespace AppBundle\Controller;
+
+        use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+        use Symfony\Component\Routing\Annotation\Route;
+
+        class MainController extends Controller
```

Comment:
Why not extending AbstractController here which was introduced in 3.3?

Reply:
You are right ... but in 3.4 we keep extending from `Controller`. In 4.2 and master we extend from `AbstractController`.